src/Angular.js

```diff
@@ -1129,9 +1129,9 @@ function parseKeyValue(/**string*/keyValue) {
   forEach((keyValue || "").split('&'), function(keyValue) {
     if ( keyValue ) {
       key_value = keyValue.replace(/\+/g,'%20').split('=');
```

Comment:
I think it will be nicer just to use `indexOf`:

```js
  var key, value, splitPoint;
  ...
    if ( keyValue ) {
      key = keyValue = keyValue.replace(/\+/g,'%20');

      splitPoint = keyValue.indexOf('=');
      if (splitPoint != -1) {
        key = keyValue.substring(0, splitPoint);
        value = keyValue.substring(splitPoint+1);
      }

      key = tryDecodeURIComponent(key);
      if ( isDefined(key) ) {

        value = tryDecodeURIComponent(value) || true;
        if (!hasOwnProperty.call(obj, key)) {
          obj[key] = val;
        } else if(isArray(obj[key])) {
          obj[key].push(val);
        } else {
          obj[key] = [obj[key],val];
        }
      }
    }
```